### DISCUSSION TITLE
can the generic argument passed to defineProps be an imported type ?

### DISCUSSION THREAD

**Discussion by chakAs3** [vuejs/language-tools#2549] [General]
i have seen i vue-meta-compoent tests workplaces 

```ts
<script setup lang="ts">
import { MyProps } from './my-props';

withDefaults(defineProps<MyProps>(), {
	bar: 1,
	baz: () => ['foo', 'bar'],
});
</script>
```

is it working in the current stable version of vue 3.2.47
as i know The interface or object literal type can contain references to types imported from other files, however, the generic argument itself passed to defineProps cannot be an imported type

**Comment by johnsoncodehk** [comment]
It would work for vue 3.3, FYI: https://twitter.com/youyuxi/status/1646812160360869889
